var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -207,6 +207,10 @@ def setup_dependent_package(self, module, dep_spec):
         self.spec.mpicxx = join_path(self.prefix.bin, 'mpicxx')
         self.spec.mpifc  = join_path(self.prefix.bin, 'mpif90')
         self.spec.mpif77 = join_path(self.prefix.bin, 'mpif77')
+        if '+slurm' in self.spec:
+            self.spec.run = join_path(self.prefix.bin, 'srun')
```

Comment:
How do you see setting a module attribute for `mpirun` ?
```python
if '+slurm' in self.spec:
    module.mpirun = which('srun)
else
    module.mpirun = Executable(join_path(self.prefix.bin, 'mpirun'))
```
Client code could then just use `mpirun(parameters)` for the tests